src/toast/toast.test.tsx

```diff
@@ -0,0 +1,219 @@
+    );
+
+    render(<ButtonWithToast />, {wrapper});
+
+    const button = await screen.findByText(displayToastButtonText);
+
+    userEvent.click(button);
+
+    expect(await screen.findByText(testToastContentText)).toBeVisible();
+
+    await waitFor(
+      () => {
+        expect(screen.queryByText(testToastContentText)).toBeInTheDocument();
+      },
+      {timeout: DEFAULT_TOAST_TIMEOUT}
+    );
+  });
+
+  it("displays at most 3 toasts", async () => {
+    const limit = 3;
+    const wrapper = ({children}: {children: React.ReactNode}) => (
+      <ToastContextProvider autoCloseToasts={false} limit={limit}>
+        {children}
+      </ToastContextProvider>
+    );
+
+    render(<ButtonWithToast numberOfToasts={5} />, {wrapper});
+
+    const button = await screen.findByText(displayToastButtonText);
+
+    userEvent.click(button);
+
+    const elements = await screen.findAllByText(testToastContentText);
+
+    expect(elements.length).toBe(limit);
+  });
+
+  it("all toasts disappears after 200ms", async () => {
+    const timeout = 200;
+
+    const wrapper = ({children}: {children: React.ReactNode}) => (
+      <ToastContextProvider defaultAutoCloseTimeout={timeout}>
+        {children}
+      </ToastContextProvider>
+    );
+
+    render(<ButtonWithToast />, {wrapper});
+
+    const button = await screen.findByText(displayToastButtonText);
+
+    userEvent.click(button);
+
+    expect(await screen.findByText(testToastContentText)).toBeVisible();
+
+    await waitFor(
+      () => {
+        expect(screen.queryByText(testToastContentText)).toBeInTheDocument();
+      },
+      {timeout}
+    );
+  });
+
+  it("autocloses itself in 2000ms", async () => {
+    const timeout = 2000;
+    const wrapper = ({children}: {children: React.ReactNode}) => (
+      <ToastContextProvider>{children}</ToastContextProvider>
+    );
+
+    render(<ButtonWithToast autoClose={true} timeout={timeout} />, {wrapper});
+
+    const button = await screen.findByText(displayToastButtonText);
+
+    userEvent.click(button);
+
+    expect(await screen.findByText(testToastContentText)).toBeVisible();
+
+    await waitFor(
+      () => {
+        expect(screen.queryByText(testToastContentText)).not.toBeInTheDocument();
+      },
+      {timeout}
+    );
+  });
+
+  it("hides toast with specific id", async () => {
+    const wrapper = ({children}: {children: React.ReactNode}) => (
+      <ToastContextProvider autoCloseToasts={false}>{children}</ToastContextProvider>
+    );
+
+    render(<ButtonWithToast />, {wrapper});
+
+    const displayButton = await screen.findByText(displayToastButtonText);
+
+    userEvent.click(displayButton);
+
+    expect(await screen.findByText(testToastContentText)).toBeVisible();
+
+    const hideButton = await screen.findByText(hideToastButtonText);
+
+    userEvent.click(hideButton);
+
+    expect(screen.queryByText(testToastContentText)).not.toBeInTheDocument();
+  });
+
+  it("Displays toast with close button", async () => {
```

Comment:
do we have a convention about case messages? maybe we can say all of them should start with lowercase